tools/bazar/controllers/ListController.php

```diff
@@ -11,57 +13,54 @@ class ListController extends YesWikiController
-
-    public function __construct(ListManager $listManager, SecurityController $securityController)
-    {
+    protected $aclService;
+    protected $authController;
+
+    public function __construct(
+        ListManager $listManager,
+        SecurityController $securityController,
+        AclService $aclService,
+        AuthController $authController
+    ) {
         $this->listManager = $listManager;
         $this->securityController = $securityController;
+        $this->aclService = $aclService;
+        $this->authController = $authController;
     }
 
     public function displayAll()
     {
         if (isset($_POST['imported-list'])) {
             foreach ($_POST['imported-list'] as $listRaw) {
                 $list = json_decode($listRaw, true);
-                $this->listManager->create($list['titre_liste'], $list['label']);
+                $this->listManager->create($list['title'], $list['nodes']);
             }
             echo '<div class="alert alert-success">' . _t('BAZ_LIST_IMPORT_SUCCESSFULL') . '.</div>';
+            echo '<div class="alert alert-success">' . _t('BAZ_LIST_IMPORT_SUCCESSFULL') . '.</div>';
         }
 
         $lists = $this->listManager->getAll();
 
-        $values = [];
         foreach ($lists as $key => $list) {
-            $values[$key]['title'] = $list['titre_liste'];
-            $values[$key]['options'] = $list['label'];
-            $values[$key]['canEdit'] = !$this->securityController->isWikiHibernated() && $this->wiki->HasAccess('write', $key);
-            $values[$key]['canDelete'] = !$this->securityController->isWikiHibernated() && ($this->wiki->UserIsAdmin() || $this->wiki->UserIsOwner($key));
+            $lists[$key]['canEdit'] = !$this->securityController->isWikiHibernated() && $this->wiki->HasAccess('write', $key);
+            $lists[$key]['canDelete'] = !$this->securityController->isWikiHibernated() && ($this->wiki->UserIsAdmin() || $this->wiki->UserIsOwner($key));
         }
 
         return $this->render('@bazar/lists/list_table.twig', [
-            'lists' => $values,
-            'loggedUser' => $this->wiki->GetUser(),
+            'lists' => $lists,
+            'loggedUser' => $this->authController->getLoggedUser(),
             'canCreate' => !$this->securityController->isWikiHibernated(),
         ]);
     }
 
     public function create()
     {
-        if (isset($_POST['valider'])) {
-            $i = 1;
-            $values = [];
-            foreach ($_POST['label'] as $label) {
-                if (($label != null || $label != '') && ($_POST['id'][$i] != null || $_POST['id'][$i] != '')) {
-                    $values[$_POST['id'][$i]] = $label;
-                    $i++;
-                }
-            }
-
-            $listeId = $this->listManager->create($_POST['titre_liste'], $values);
+        if (isset($_POST['submit'])) {
+            $listeId = $this->listManager->create($_POST['title'], json_decode($_POST['nodes'], true));
```

Comment:
should we check if user has the rights to create a list ?

Reply:
the list creation is currently only disabled when wiki is "hibernated"
Do you want to change that? (the check is made in `ListController->displayAll` and `ListManager->create`

Reply:
if it's checked in the manager, it's ok, i suppose i just thought that COntroller should control :D

In any case, we can change it later, while opening to non admins users